.github/workflows/build_and_cache_Kokkos_linux.yml

```diff
@@ -0,0 +1,112 @@
+name: Build and Cache Kokkos and Kokkos Kernels
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        required: true
+        type: string
+    outputs:
+      exec_model:
+        description: "The execution model for Kokkos."
+        value: ${{ jobs.linux-set-builder-matrix.outputs.exec_model }}
+      kokkos_version:
+        description: "Kokkos version"
+        value: ${{ jobs.linux-set-builder-matrix.outputs.kokkos_version }}
+
+jobs:
+  linux-set-builder-matrix:
+    name: Set Kokkos builder matrix
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.10.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Kokkos execution strategy
+        id: exec_model
+        run: echo "::set-output name=exec_model::[\"SERIAL\"]" # We may also adopt [OPENMP, THREADS] in later iterations
```

Comment:
Can't we just set this directly in matrix? I think it is easier to see what's going on.

Reply:
These parameters are set here and used as an output of this job.
This way, we can have consistent access to this information around our CI/CD pipelines.

Reply:
Unfortunately this is the recommended way to build these types of workflows where one depends on another.
The evaluation order means that the matrix cannot be used for these types of jobs.

Reply:
I understand why it is required for python versions (it distinguishes pull_request and `master` merge), but I wasn't sure it is also the case for this workflow. Still, if it is the simplest way to do that, I am okay.